test/unit/factories/CanonGuardFactory.t.sol

```diff
@@ -88,5 +90,8 @@ contract UnitCanonGuardFactory is Test {
 
     // it should store the contract as a factory children
     assertTrue(canonGuardFactory.isChild(_canonGuard));
+
+    // it should match the deterministic address
+    assertEq(_canonGuard, CREATE3.predictDeterministicAddress(keccak256(abi.encode(_safe)), address(canonGuardFactory)));
```

Comment:
I don't get how u are only passing the safe in the abi.encode and it's generating the same address